tst/ModelTest.php

```diff
@@ -102,6 +102,58 @@ public function testBasicWorkflow()
         $this->assertEquals(array(), $paste->getComments(), 'comment was deleted with paste');
     }
 
+    public function testPasteV1()
+    {
+        $pasteData = Helper::getPaste(1);
+        unset($pasteData['meta']['formatter']);
+
+        $path = $this->_path . DIRECTORY_SEPARATOR . 'v1-test.sq3';
+        if (is_file($path)) {
+            unlink($path);
+        }
+        $options                   = parse_ini_file(CONF_SAMPLE, true);
+        $options['purge']['limit'] = 0;
+        $options['model']          = array(
+            'class' => 'Database',
+        );
+        $options['model_options'] = array(
+            'dsn' => 'sqlite:' . $path,
+            'usr' => null,
+            'pwd' => null,
+            'opt' => array(PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION),
+        );
+        Helper::createIniFile(CONF, $options);
+        $model = new Model(new Configuration);
+        $model->getPaste('0000000000000000')->exists(); // triggers database table creation
+        $model->getPaste(Helper::getPasteId())->delete(); // deletes the cache
+
+        $db = new PDO(
+            $options['model_options']['dsn'],
+            $options['model_options']['usr'],
+            $options['model_options']['pwd'],
+            $options['model_options']['opt']
+        );
+        $statement = $db->prepare('INSERT INTO paste VALUES(?,?,?,?,?,?,?,?,?)');
+        $statement->execute(
+            array(
+                Helper::getPasteId(),
+                $pasteData['data'],
+                $pasteData['meta']['postdate'],
+                0,
+                0,
+                0,
+                json_encode($pasteData['meta']),
+                null,
+                null,
+            )
+        );
+        $statement->closeCursor();
+
+        $paste = $model->getPaste(Helper::getPasteId());
+        $paste->getDeleteToken();
```

Comment:
Do I miss sth. or why do you get the delete token here, but do not use the return value whatsoever?

So either don't get it or test the deletion actually. (which would maybe be a new test, however)

Reply:
I probably added the line here to get coverage on the condition, that calls a get(), if the paste didn't get loaded yet. I could add a separate test to cover the same call separately, but the purpose here isn't to test that `getDeleteToken()`returns anything, just that calling it will cause the loading of the paste from storage.
https://github.com/PrivateBin/PrivateBin/blob/af54e70359ae5994abc9f505e33cea6fdccf639c/lib/Model/Paste.php#L177-L179

Reply:
…but if the remove the line the test suceeds anyway, does not it?

If so, then IMHO that's not a good test. It should fail if one does not do the thing it tries to assert…

Reply:
I don't see a way to get that section of code covered with any kind of assert and separating it would increase the test run cost (duplicating every part till that line), but be even less clear, wouldn't it? For reference, here it is in the public coverage report on the latest release:
https://privatebin.info/coverage/Model/Paste.php.html#178

Reply:
I did move it into an assertNotEmpty, mainly so there is a comment explaining what it is for.